### DISCUSSION TITLE
How to use complex proof restrictions using wql

### DISCUSSION THREAD

**Discussion by sschulz-t** [openwallet-foundation/credo-ts#1255] [Q&A] [answered]
Is there a way to request a proof that is using a "complex" restrictions for an attribute?

I have a use case where I want to send complex queries using the indy wql (https://github.com/hyperledger/indy-sdk/tree/main/docs/design/011-wallet-query-language).

For example I want a proof that the person has either membership a or membership b in one proof request.
Example code that works on the low level indy-sdk during proof creation:
```
"attr1_referent": {
        "restrictions": [
          {
            "cred_def_id": {
              "$in": [
                "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx:3:CL:129:MEMBERSHIP_A",
                "yyyyyyyyyyyyyyyyyyyyyyyyyyy:3:CL:129:MEMBERSHIP_B"
              ]
            }
          }
        ],
        "name": "last_name"
      },
}
```
How can I do this request with this framework?

**Comment by TimoGlastra** [comment, marked as answer]
Hi @sschulz-t, 

I never knew this syntax was supported, and I've never seen anyone use it. I did some small modifications in AFJ to disable the current validation of requiring `cred_def_id` to be a string, and to my surprise the `$in` statement works. 

For now, as this will fail validation in AFJ you can use the following syntax:

```
"attr1_referent": {
        "restrictions": [
          {
            "cred_def_id": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx:3:CL:129:MEMBERSHIP_A"
          },
          {
            "cred_def_id": "yyyyyyyyyyyyyyyyyyyyyyyyyyy:3:CL:129:MEMBERSHIP_B"
          },
        ],
        "name": "last_name"
      },
}
```

@swcurran interesting to take into account for the AnonCreds Spec. Apparently the value for the restrictions can be complex WQL queries. Both ACA-Py and AFJ require it to be a string, but technically the Indy SDK supports this.

**Reply by sschulz-t** [reply]
Thanks for the fast feedback and the workaround. So apparently the list of restrictions is interpreted as "OR", good to know :) Seems to work for my use case.
Not sure if anyone is actually using the complex wsl queries for the restrictions, they just came in handy for my use case :)
